components/plugin_marketplace/marketplace_list/navigation_row/navigation_button.tsx

```diff
@@ -2,22 +2,21 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import PropTypes from 'prop-types';
 import {FormattedMessage} from 'react-intl';
 
-export default class NavigationButton extends React.PureComponent {
-    static propTypes = {
-        onClick: PropTypes.func.isRequired,
-        messageId: PropTypes.string.isRequired,
-        defaultMessage: PropTypes.string.isRequired,
-    };
+type NavigationButtonProps = {
+    onClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void,
+    messageId: string,
+    defaultMessage: string,
+};
 
-    onClick = (e) => {
-        e.preventDefault();
-        this.props.onClick();
+export default class NavigationButton extends React.PureComponent <NavigationButtonProps> {
+    onClick = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>):void => {
+        event.preventDefault();
+        this.props.onClick(event);
     };
 
-    render() {
+    render():JSX.Element {
```

Comment:
Nit, can we add whitespace between the colon and type in all instances in this PR?

```suggestion
    render(): JSX.Element {
```